internal/framework/status/updater.go

```diff
@@ -169,3 +182,43 @@ func (upd *updaterImpl) update(
 			"kind", obj.GetObjectKind().GroupVersionKind().Kind)
 	}
 }
+
+func UpdateObjStatus(
+	ctx context.Context,
+	attempts int,
+	sleep time.Duration,
+	obj client.Object,
+	upd *UpdaterImpl,
+) (err error) {
+	for i := 0; i < attempts; i++ {
+		if i > 0 {
+			time.Sleep(sleep)
+			sleep *= 2
+		}
+		if err = upd.cfg.Client.Status().Update(ctx, obj); err == nil {
+			return nil
+		}
+	}
+	return err
+}
+
+func GetObj(
+	ctx context.Context,
+	attempts int,
+	sleep time.Duration,
+	obj client.Object,
+	upd *UpdaterImpl,
+	nsname types.NamespacedName,
+) (err error) {
+	for i := 0; i < attempts; i++ {
```

Comment:
GetObj and UpdateObj look really similar. Can we refactor to just one function that a retries a function with exponential backoff? Then you can pass in the function you want to retry as an argument. It would look something like this:
```
func retryWithExpBackoff(
        ctx context.Context,
	attempts int,
	sleep time.Duration,
        operation func(ctx) error,
) { ...}
```

and then when you call it:
```
getOperation := func(ctx) error {
    if err = upd.cfg.Client.Get(ctx, nsname, obj); err == nil || apierrors.IsNotFound(err) {
			return nil
    }
    return err
}

retryWithExpBackoff(ctx, attempts, sleep, getOperation)
```

Reply:
If we write it this way we could also re-use it in other areas of the code.

Reply:
We could also pull in a package to do the backoff retry so we don't have to maintain it: https://github.com/cenkalti/backoff

Reply:
@pleshakov @sjberman @ciarams87 thoughts on pulling in a package for exponential backoff?

Reply:
This is simple enough where I'd rather not add another dependency.

Reply:
there is also this function than looks useful https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#ExponentialBackoffWithContext 
we already use https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#PollUntilContextCancel in https://github.com/nginxinc/nginx-kubernetes-gateway/blob/bddcc0df74c9f3ac980f48fb2ac90dd025667ff3/internal/mode/static/nginx/runtime/manager.go#L146

Reply:
Good find @pleshakov. I like the idea of using that